modules/ximgproc/src/paillou_filter.cpp

```diff
@@ -0,0 +1,454 @@
+				c1 = (short*)img.ptr(rows - 1) + j;
+				border = *c1;
+				ym[rows - 1] = *c1;
+				c1 -= cols;
+				ym[rows - 2] = *c1 - b1*ym[rows - 1];
+				c1 -= cols;
+				for (int i = rows - 3; i >= 0; i--, c1 -= cols)
+					ym[i] = *c1 - b1*ym[i + 1] - b2*ym[i + 2];
+				// Formule 25 p193
+				for (int i = 0; i<rows; i++, f2 += cols)
+					*f2 = (float)(a1*(ym[i] - yp[i]));
+			}
+		}
+		break;
+		case CV_16U :
+        {
+			for (int j = range.start; j<range.end; j++)
+			{
+				// Formule 26 p194
+				unsigned short *c1 = (unsigned short*)img.ptr(0) + j;// c1 pointeur sur x(i-1)
+				f2 = ((float*)im1.ptr(0)) + j;
+				double border = *c1;
+				yp[0] = *c1;
+				c1 += cols;
+				yp[1] = *c1 - b1*yp[0] - b2*border;
+				c1 += cols;
+				for (int i = 2; i<rows; i++, c1 += cols)
+					yp[i] = *c1 - b1*yp[i - 1] - b2*yp[i - 2];
+				// Formule 27 p194
+				c1 = (unsigned short*)img.ptr(rows - 1) + j;
+				border = *c1;
+				ym[rows - 1] = *c1;
+				c1 -= cols;
+				ym[rows - 2] = *c1 - b1*ym[rows - 1];
+				c1 -= cols;
+				for (int i = rows - 3; i >= 0; i--, c1 -= cols)
+					ym[i] = *c1 - b1*ym[i + 1] - b2*ym[i + 2];
+				// Formule 25 p193
+				for (int i = 0; i<rows; i++, f2 += cols)
+					*f2 = (float)(a1*(ym[i] - yp[i]));
+			}
+		}
+            break;
+        case CV_32S :
+        case CV_32F :
+        case CV_64F :
+        default :
+            delete []ym;
+            delete []yp;
+            return ;
+            }
+        delete []ym;
+        delete []yp;
+    };
+    ParallelGradientPaillouYCols& operator=(const ParallelGradientPaillouYCols &) {
+         return *this;
+    };
+};
+
+
+class ParallelGradientPaillouYRows: public ParallelLoopBody
+{
+private:
+    Mat &img;
+    Mat &dst;
+    double a;
+    double w;
+    bool verbose;
+
+public:
+    ParallelGradientPaillouYRows(Mat& imgSrc, Mat &d,double aa,double ww):
+        img(imgSrc),
+        dst(d),
+        a(aa),
+        w(ww),
+        verbose(false)
+    {}
+    void Verbose(bool b){verbose=b;}
+    virtual void operator()(const Range& range) const
+    {
+        if (verbose)
+            std::cout << getThreadNum()<<"# :Start from row " << range.start << " to "  << range.end-1<<" ("<<range.end-range.start<<" loops)" << std::endl;
+        float *iy,*iy0;
+        int tailleSequence=(img.rows>img.cols)?img.rows:img.cols;
+        double *iym=new double[tailleSequence],*iyp=new double[tailleSequence];
+        int cols=img.cols;
+
+        // Formule 13
+        double				d=(1-2*exp(-a)*cosh(w)+exp(-2*a))/(2*a*exp(-a)*sinh(w)+w*(1-exp(-2*a)));
+        double				c1=a*d;
+        double				c2=w*d;
+        // formule 12
+        double				b1=-2*exp(-a)*cosh(w);
+        double				b2=exp(-2*a);
+        // Formule 14
+        double				a0p=c2;
+        double				a1p=(c1*sinh(w)-c2*cosh(w))*exp(-a);
+        double				a1m=a1p-c2*b1;
+        double				a2m=-c2*b2;
+
+        for (int i=range.start;i<range.end;i++)
+            {
+            iy0 = ((float*)img.ptr(i));
```

Comment:
Thanks you for comment. When you think comment are finished tell me and I will try to insert all comment in a commit. I think that  for PR 701 comments are same may be it's not usefull to check it. I will push a new commit too